### ISSUE TITLE
3ds_injector needs updating

### ISSUE THREAD

**Issue by Madridi7** [yifanlu/3ds_injector#7]
Cosmo3ds is not compiling, my understanding is that it's an injector issue which needs updating. I think it needs to be compatible with latest ctrulib

**Comment by profi200** [comment]
"needs updating" and "i think" is not helpful at all. Provide build logs.

**Comment by sykhro** [comment]
I'm just going to guess and say he is probably referring to this not building anymore after smealum/ctrulib@cf91be5

**Comment by Madridi7** [comment]
I deliberately did not add a build log as I'm using a slightly older ctrulib (2-3 commits back), since the latest ctrulib broke a lot of other stuff. But I do have confirmation from other users it's still broken with latest ctrulib. Anyway, here is the log:

`make[1]: Entering directory`/c/Compiling/Cosmo3DS/injector'
fsldr.c
arm-none-eabi-gcc -MMD -MP -MF /c/Compiling/Cosmo3DS/injector/build/fsldr.d -flt
o -Wall -Os -mword-relocations -fomit-frame-pointer -ffunction-sections -fdata-s
ections -march=armv6k -mtune=mpcore -mfloat-abi=hard -mtp=soft -I/c/Compiling/Co
smo3DS/injector/include -I/c/devkitpro/libctru/include -I/c/Compiling/Cosmo3DS/i
njector/build -DARM11 -D_3DS -c /c/Compiling/Cosmo3DS/injector/source/fsldr.c -o
 fsldr.o
c:/Compiling/Cosmo3DS/injector/source/fsldr.c: In function 'FSLDR_OpenFileDirect
ly':
c:/Compiling/Cosmo3DS/injector/source/fsldr.c:91:22: error: request for member '
id' in something not a structure or union
   cmdbuf[2] = archive.id;
                      ^
c:/Compiling/Cosmo3DS/injector/source/fsldr.c:92:22: error: request for member '
lowPath' in something not a structure or union
   cmdbuf[3] = archive.lowPath.type;
                      ^
c:/Compiling/Cosmo3DS/injector/source/fsldr.c:93:22: error: request for member '
lowPath' in something not a structure or union
   cmdbuf[4] = archive.lowPath.size;
                      ^
c:/Compiling/Cosmo3DS/injector/source/fsldr.c:98:44: error: request for member '
lowPath' in something not a structure or union
   cmdbuf[9] = IPC_Desc_StaticBuffer(archive.lowPath.size, 2);
                                            ^
c:/Compiling/Cosmo3DS/injector/source/fsldr.c:99:29: error: request for member '
lowPath' in something not a structure or union
   cmdbuf[10] = (u32) archive.lowPath.data;
                             ^
make[2]: **\* [fsldr.o] Error 1
make[1]: **\* [build] Error 2
make[1]: Leaving directory `/c/Compiling/Cosmo3DS/injector'
make: *** [build/patches.h] Error 2`

**Comment by profi200** [comment]
Yeah, this is because of the changes @Steveice10 made to fs:USER. It needs adjustments.

**Comment by Madridi7** [comment]
Will this get fixed? Can't compile Cosmo3DS without it

**Comment by yifanlu** [comment]
Someone should fix it and submit a pull request because I don't have the time.
